Fixes for Trixi.jl

benegee opened the pull request.

JoshuaLampert commented on src/TrixiTest.jl:
```diff
@@ -1,6 +1,7 @@
 
-using Test: @test, @testset
+using Test: @inferred, @test, @test_broken, @test_deprecated, @test_logs,
+            @test_nowarn, @test_skip, @test_throws, @test_warn, @testset
```
It feels a bit strange to import all of these macros explicitly even though they are not needed here. I know we somehow need this in order to access them from the tests in Trixi.jl, but would this also work with a simple `using Test`?

JoshuaLampert reviewed (commented):
Thanks for all the effort. I honestly can't judge if moving these `esc` around is the correct way to do this, but if this fixes things, it's nice. IIUC, this fixes some issues that came up in the PR to Trixi.jl. Can we add tests here for the things, which failed in Trixi.jl before, but should be fixed by this PR?

JoshuaLampert commented on src/macros.jl:
```diff
@@ -75,20 +74,23 @@ macro test_trixi_include_base(elixir, args...)
+        args = append_to_kwargs(args, :additional_ignore_content,
+                                [
+                                    r"┌ Warning: Interrupted\. Larger maxiters is needed\..*\n└ @ SciMLBase .+\n",
+                                    r"┌ Warning: Interrupted\. Larger maxiters is needed\..*\n└ @ Trixi .+\n"
```
I know this has been there before, but I just saw this. Does it make sense to have a Trixi.jl specific ignore content in TrixiTest.jl? Could we also add this to the ignore content from Trixi.jl? Maybe in `@test_trixi_include` with something lik
```julia
if !isnothing(get_kwarg(args, :maxiters))
	args = append_to_kwargs(args, :additional_ignore_content,
                                [r"┌ Warning: Interrupted\. Larger maxiters is needed\..*\n└ @ Trixi .+\n"])
end
```
(but maybe it's also just simpler to keep it here)

benegee replied:
Yes, absolutely.

benegee commented:
> What's the status of this PR?

This escalated a bit. My initial goal was to be able to use pre computed error bounds, like
```julia
l2_error = 1e-6
@test_trixi_include(elixir, l2 = l2_error)
```

While this was possible after a while, something like
```julia
new_x = 5
@test_trixi_include(elixir, x = new_x)
```
would still not work. Then I realized, we already use `global` in such situations
https://github.com/trixi-framework/Trixi.jl/blob/9acf23d989667880c7690990852c5cdf52ecff59/test/test_dgmulti_1d.jl#L138
and this work for `l2` as well, so I reverted most of the changes.

> I honestly can't judge if moving these `esc` around is the correct way to do this, but if this fixes things, it's nice.

I saw this pattern often in discourse and SO discussions when the goal was to avoid macro hygiene s.t. objects in outer scope would still be accessible. Anyway, I now reverted to the previous state almost everywhere. 

> IIUC, this fixes some issues that came up in the PR to Trixi.jl. Can we add tests here for the things, which failed in Trixi.jl before, but should be fixed by this PR?

Ultimately, only additional macros from Test.jl are required, or the whole Test.jl package. Everything else need to be fixed in Trixi.jl.
